modules/account-lib/src/coin/eth/utils.ts

```diff
@@ -260,12 +260,12 @@ export function decodeTokenTransferData(data: string): TokenTransferData {
   );
 
   return {
-    to: addHexPrefix(to),
+    to: addHexPrefix(to.toString('hex')),
```

Comment:
is this not already a hex string?
same question for other conversions below

Reply:
No this is a buffer. We need to convert it to string and its true for all conversions below.